streams/src/main/java/org/apache/kafka/streams/state/Stores.java

```diff
@@ -215,6 +218,16 @@ public static WindowBytesStoreSupplier persistentWindowStore(final String name,
         if (segmentInterval < 60_000) {
             throw new IllegalArgumentException("segmentInterval must be at least one minute");
         }
+        if (windowSize > retentionPeriod) {
+            throw new IllegalArgumentException("The retention period of the window store "
+                                                   + name + " must be no smaller than its window size. Got size=["
+                                                   + windowSize + "], retention=[" + retentionPeriod + "]");
+        }
```

Comment:
I've added this check since I removed it from the DSL builder.

Reply:
nit: fix indention

Reply:
This indentation is actually "correct" according to the IDEA defaults (a continuation of string concatenation gets indented).

Is that ok with you, or do you want me to update my formatter?

Reply:
We usually don't indent but align -- at least, all other code is formatter like this IIRC. But it's a nit anyway...